src/object/srv_obj_migrate.c

```diff
@@ -1220,9 +1227,42 @@ migrate_one_ult(void *arg)
 		goto out;
 	}
 
-	rc = migrate_dkey(tls, mrone);
-	D_DEBUG(DB_REBUILD, DF_UOID" migrate dkey "DF_KEY" rc %d\n",
-		DP_UOID(mrone->mo_oid), DP_KEY(&mrone->mo_dkey), rc);
+	data_size = daos_iods_len(mrone->mo_iods, mrone->mo_iod_num);
+	D_DEBUG(DB_TRACE, "mrone %p data size is "DF_U64"\n",
+		mrone, data_size);
+	if (data_size == 0) {
+		D_DEBUG(DB_REBUILD, "skipe empty mrone %p\n", mrone);
+		D_GOTO(out, rc);
+	}
+
+	D_DEBUG(DB_REBUILD, "mrone %p inflight iod %d max iod %d\n",
+		mrone, tls->mpt_inflight_iod, tls->mpt_inflight_max_iod);
+
+	while (tls->mpt_inflight_iod + mrone->mo_iod_num >=
+	       tls->mpt_inflight_max_iod && tls->mpt_inflight_max_iod != 0
+	       && !tls->mpt_fini) {
```

Comment:
may consider to only create migrate_one_ult when not-busy case later, as currently still create all ULTs one time that may cause ABT scheduling pressure.

Reply:
yes, the new update will control ULT count as well here.

Reply:
I agree with Xuehzao. The wait condition check of max size and max ULT count should be put together.

Reply:
Hmm, we do not know the size when creating ULT to enumerate the object.

Reply:
but you know the per-xstream inflight ULTs here, right?